cni/pkg/install-cni/pkg/install/kubeconfig_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/coreos/etcd/pkg/fileutil"
+	"github.com/stretchr/testify/assert"
+
+	"istio.io/istio/cni/pkg/install-cni/pkg/config"
+	testutils "istio.io/istio/pilot/test/util"
+)
+
+const (
+	kubeconfigMode = 0600
+	k8sServiceHost = "10.96.0.1"
+	k8sServicePort = "443"
+	saToken        = "service_account_token_string"
+)
+
+func TestCreateKubeconfigFile(t *testing.T) {
+	saDir, err := ioutil.TempDir("", "serviceaccount-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := os.RemoveAll(saDir); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	data, err := ioutil.ReadFile(filepath.Join("testdata", "kube-ca.crt"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubeCAFilepath := filepath.Join(saDir, "kube-ca.crt")
+	err = ioutil.WriteFile(kubeCAFilepath, data, 0644)
```

Comment:
Is this file mode correct? If so, should it be in the constants file?

Reply:
Ended up removing this file copy. It wasn't necessary.